hack/jenkins/benchmark-dockerized.sh

```diff
@@ -0,0 +1,53 @@
+#!/bin/bash
+
+# Copyright 2018 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+set -o xtrace
+
+retry() {
+  for i in {1..5}; do
+    "$@" && return 0 || sleep $i
+  done
+  "$@"
+}
+
+# Runs benchmark integration tests, producing JUnit-style XML test
+# reports in ${WORKSPACE}/artifacts. This script is intended to be run from
+# kubekins-test container with a kubernetes repo mapped in. See
+# k8s.io/test-infra/scenarios/kubernetes_verify.py
+
+export PATH=${GOPATH}/bin:${PWD}/third_party/etcd:/usr/local/go/bin:${PATH}
+
+retry go get github.com/tools/godep && godep version
+retry go get github.com/jstemmer/go-junit-report
+
+# Disable the Go race detector.
+export KUBE_RACE=" "
```

Comment:
Why?

Reply:
In general I've seen (from e.g.s on the internet) that the go race detector is not used with benchmarks.
I read up a bit about it and I believe it's probably because it affects the real performance of the benchmark due to adding extra instrumentation to for e.g memory accesses. Also some extra runtime libraries are enabled to watch accesses to shared variables. It's explained here - https://blog.golang.org/race-detector

Also I found this about the runtime overhead - https://golang.org/doc/articles/race_detector.html#Runtime_Overheads (which imo we shouldn't be adding for benchmarks).
WDYT?

Reply:
Sorry, I wasn't clear. I agree we definitely shouldn't have race detector switched on for benchmarks, because it affects performance.
My question was "isn't it off by default"?

Reply:
I guess so. Though it seems to be explicitly set at places (e.g https://github.com/kubernetes/kubernetes/blob/master/hack/make-rules/test-integration.sh#L75, https://github.com/kubernetes/kubernetes/blob/master/hack/benchmark-go.sh#L26) - probably to be safe (in case the env var is pre-set).